kratos/spatial_containers/octree_binary.h

```diff
@@ -120,7 +120,68 @@ namespace Kratos {
         virtual ~OctreeBinary() {
             delete root_;
         }
-        
+
+        ///@}
+        ///@name Operators
+        ///@{
+
+
+        ///@}
+        ///@name Operations
+        ///@{
+
+        /**
+         * @brief This method sets manually the scale factors of the BB
+         * @param NewScaleFactor The scalar factors to be set
+         */
+        void SetScaleFactor(const double*  NewScaleFactor)
+        {
+            for(int i = 0 ; i < DIMENSION ; i++) {
+                mScaleFactor[i] = NewScaleFactor[i];
+            }
+        }
+
+        /**
+         * @brief This method sets manually the offsets of the BB
+         * @param NewOffset The offsets to be set
+         */
+        void SetOffset(const double*  NewOffset)
+        {
+            for(int i = 0 ; i < DIMENSION ; i++) {
+                mOffset[i] = NewOffset[i];
+            }
+        }
+
+        /**
+         * @brief This method gets the scale factors of the BB
+         * @return The scalar factors of the octree
+         */
+        double* GetScaleFactor()
```

Comment:
This does not look safe... Why do need to do it? Can you not return a unique pointer for example?

Reply:
I am using a C array because is the system used in the tree, in my opinion would be better to use an array_1d

Reply:
In any case its is probably safer to mark the method `const` (or return as `const` without making a copy)

Reply:
Done